src/components/ProfilePicPopup.js

```diff
@@ -0,0 +1,83 @@
+import Popup from "./Popup";
+
+class ProfilePicPopup extends Popup {
+  constructor({ popupSelector, avatarSelector, handleFormSubmit }) {
+    super(popupSelector);
+    this._popupForm = this.popupElement.querySelector(".popup__form");
+    this._submitButton = this._popupForm.querySelector(".popup__button");
+    this._defaultButtonText = this._submitButton.textContent;
+    this._avatarElement = document.querySelector(avatarSelector);
+    this._handleFormSubmit = handleFormSubmit;
+    this._inputList = Array.from(
+      this._popupForm.querySelectorAll(".popup__input")
+    );
+    this._setEventListeners();
+  }
+
+  _getInputValues() {
+    const userInput = this._inputList.reduce((acc, input) => {
+      const { name, value } = input;
+      acc[name] = value;
+      return acc;
+    }, {});
+    return userInput;
+  }
+
```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```

Reply:
Did not understand this comment here.